File: C++/Dynamic Programming/Container With Most Water.java
//https://leetcode.com/problems/container-with-most-water/submissions/568315411/

class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int first = 0;
        int last = height.length-1;
        while(first<last){
            max = Math.max(max,Math.min(height[first],height[last])*(last-first));
            if(height[first]<height[last])
                first++;
            else
                last--;
        }
        return max;
    }
}

File: Leetcode/Container_with_most_water.cpp
//https://leetcode.com/problems/container-with-most-water/

class Solution {
public:
    int maxArea(vector<int>& h) {
        int ans = 0;
        int i = 0 , j = h.size() - 1;
        while(i < j)
        {
            ans = max(ans , min(h[i] , h[j])*(j-i));
            if(h[i] > h[j])
                j--;
            else
                i++;
        }
        return ans;
    }
};
